plutus-core/plutus-core/src/PlutusCore/Parser/ParserCommon.hs

```diff
@@ -144,25 +128,20 @@ toSrcSpan start end =
         , srcSpanECol = unPos (sourceColumn end)
         }
 
-version :: Parser (Version SourcePos)
-version = lexeme $ do
-    p <- getSourcePos
-    x <- Lex.decimal
-    void $ char '.'
-    y <- Lex.decimal
-    void $ char '.'
-    Version p x y <$> Lex.decimal
-
-version' :: Parser (Version SrcSpan)
-version' = withSpan $ \sp -> do
+version :: Parser (Version SrcSpan)
+version = withSpan $ \sp -> do
     x <- Lex.decimal
     void $ char '.'
     y <- Lex.decimal
     void $ char '.'
     Version sp x y <$> Lex.decimal
 
 name :: Parser Name
-name = lexeme $ try $ do
+name = lexeme name'
+
+-- | Like `name` but does not consume trailing whitespaces.
+name' :: Parser Name
```

Comment:
I wonder if this should be unticked and we should audit the cases where the other one is needed?

Reply:
I think the other one (which consumes whitespaces) is more common. 
- If `name` is in the middle, we do need to consume whitespaces (e.g., `LamAbs ann name body`)
- If `name` is at the end, then we don't consume whitespaces (e.g., `Var ann name`).

We can remove the other one and add `<* whitespace` whenever whitespaces need to be consumed, but I don't think it reduces the chance of forgetting to consume whitespaces.

And yes, this is ugly either way.